### PR TITLE
Add Micrometer Support for Consumer and Producer

### PR DESCRIPTION
**Pull request by crigas1**
Resolves https://github.com/reactor/reactor-kafka/issues/206#issuecomment-1006119883

cherry-picked Gary's commit (#5ad18b4), from [here](https://github.com/reactor/reactor-kafka/pull/262/commits), resolved conflicts, fixed copyrights and sinces. 
I did not port the changes around compatibility, since they are not necessary at the moment and we don't really know if and when micrometer v2 will go ahead.

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -5,6 +5,7 @@ reactorCore = "3.4.27-SNAPSHOT"
 asciidoctor = "3.3.2"
 kafka-doc = "28"
 log4j = "2.17.1"
+micrometer = "1.10.3"
```
Comment (patpatpat123) [inline gradle/libs.versions.toml:8]: Many thanks for this PR!
micrometer 1.10.4 just got released today.
Would it be possible to use the latest and greatest from this PR by bumping this 1.10.3 to 1.10.4 please?

Comment (crigas1) [reply gradle/libs.versions.toml]: addressed

Comment (patpatpat123) [reply gradle/libs.versions.toml]: Thanks a lot!

**Comment by patpatpat123** [comment]
Is this good for approval @garyrussell ?

File: src/main/java/reactor/kafka/receiver/MicrometerConsumerListener.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A consumer listener that manages {@link KafkaClientMetrics}.
+ *
+ * @author Gary Russell
+ * @since 1.3.16
```
Comment (crigas1) [inline src/main/java/reactor/kafka/receiver/MicrometerConsumerListener.java:35]: since 1.3.16 was released yesterday this needs to be updated to 1.3.17

Comment (crigas1) [reply src/main/java/reactor/kafka/receiver/MicrometerConsumerListener.java]: addressed

**Comment by crigas1** [comment]
have tested this locally, using it with one of our apps and  everything looked ok

**Review by garyrussell — APPROVED** [review]

**Comment by patpatpat123** [comment]
Thank you all

**Comment by patpatpat123** [comment]
Hey @crigas1, do you mind sharing your local test please? 

I tried the latest reactor-kafka 1.3.17-SNAPSHOT + spring-boot-starter-actuator + micrometer-registry-prometheus and unfortunately, not seeing any new / kafka-related metrics

**Comment by patpatpat123** [comment]
Many thanks @garyrussell , working now

**Pull request merged**